daemon/image_delete.go

```diff
@@ -70,7 +70,7 @@ func (daemon *Daemon) DeleteImage(eng *engine.Engine, name string, imgs *engine.
 				if parsedTag != "" {
 					tags = append(tags, parsedTag)
 				}
-			} else if repoName != parsedRepo && !force {
+			} else if repoName != parsedRepo && !force && first {
```

Comment:
`first` is the input parameter of function `DeleteImage` which will be set to true at https://github.com/docker/docker/blob/master/daemon/image_delete.go#L19 which means it is the first time to call `DeleteImage`,  since `DeleteImage` is a recursive function, `first` will be set to false at https://github.com/docker/docker/blob/master/daemon/image_delete.go#L120